client/src/components/util/useOnOutsideClick.js

```diff
@@ -12,6 +12,8 @@ export default (ref, action) => {
     return () => {
       document.removeEventListener("click", handleClickOutside);
     };
-  });
+  },
+    []
```

Comment:
Sorry, just want to understand better what's happening here. Was this listener originally unnecessary so you're just removing it to clean up the code, or how can I better understand this?

Reply:
The `useEffect` hook takes two arguments, a callback and an optional array of variables to watch for changes in order signal it's time to trigger the callback. Previously this callback was being triggered whenever React decided it ought it trigger it, by passing an empty array we're setting the callback to only trigger on mount. This is the hook equivalent to a `componentDidMount()` lifecycle method.

Reply:
Awesome, thanks for explaining!